src/harness/reference_models/aggregate_interference/aggregate_interference.py

```diff
@@ -0,0 +1,362 @@
+  interference value in dBm for each of the constraints.
+==================================================================================
+"""
+
+import numpy as np 
+import logging
+from reference_models.interference import interference as interf
+from reference_models.geo import utils
+
+def convertAndSumInterference(cbsd_interference_list):
+  """ Converts interference in dBm to mW to calculate aggregate interference """
+  
+  interferences_in_mw = [interf.dbToLinear(cbsd_interference) 
+      for cbsd_interference in cbsd_interference_list]
+
+  aggregate_interference_in_mw = np.sum(interferences_in_mw)
+
+  return interf.linearToDb(aggregate_interference_in_mw)
+
+def aggregateInterferenceForPoint(protection_point, channels, low_freq, high_freq, 
+      grant_objects, fss_info, esc_antenna_info, protection_ent_type, 
+      region_type="SUBURBAN"):
+  """Algorithm to compute aggregate interference for protection point
+  
+  This method is invoked to calculate aggregate interference for FSS(Co-channel 
+  passband) and ESC protection points.It is also invoked for protection points 
+  within PPA and GWPZ protection areas.
+
+  Args:
+   protection_point: Protection point inside protection area
+   channels: Channels over which reference incumbent needs protection
+   low_freq: Low frequency of protection constraint (Hz)
+   high_freq: High frequency of protection constraint (Hz)
+   grant_objects: A list of grant requests, each one being a namedtuple
+                  containing grant information
+   fss_info: Contains information on antenna height and weights on the tangent
+             and perpendicular components.
+   esc_antenna_info: Contains information on ESC antenna height, azimuth,
+                     gain and pattern gain 
+   protection_ent_type: An enum member of class ProtectedEntityType
+   region_type: Region type of the protection point
+   Returns:
+      Aggregate interference for a protection constraint(dBm)
+  """
+
+  aggregate_interference = {}
+
+  for channel in channels:
+    # Get protection constraint over 5MHz channel range
+    protection_constraint = interf.ProtectionConstraint(latitude=protection_point[0],
+                              longitude=protection_point[1], low_frequency=channel[0],
+                              high_frequency=channel[1], entity_type=protection_ent_type)
+
+    # Identify CBSD grants in the neighborhood of the protection point
+    # and channel
+    neighborhood_grants = interf.findOverlappingGrantsInsideNeighborhood(
+                            grant_objects, protection_constraint)
+
+    if len(neighborhood_grants) is 0:
+      continue
+
+    cbsd_interference_list = []
+    # Compute interference grant causes to protection point over channel 
+    for i, grant in enumerate(neighborhood_grants):
+
+      # Compute interference at FSS incumbent type
+      if protection_ent_type is interf.ProtectedEntityType.FSS_CO_CHANNEL or\
```

Comment:
See my comment in the IAP PR #473.
What about delegating the switch to a top level function in the interf module ?

Reply:
Calling   entry level function from interference module